### PR TITLE
Add property tests

### PR DESCRIPTION
**Pull request by andrewh**
## Summary

- Add property-based testing across the codebase using `pgregory.net/rapid`
- Fix two bugs in `MarshalConfig` found by property testing: traffic rate overflow and fractional rate format

### Import pipeline (16 tests)

- **Tree construction**: all spans present, single root, full reachability, no cycles
- **Stats collection**: counts match input, errors bounded, durations length/positivity, call counts match tree children
- **Duration arithmetic**: mean in [min,max], idempotence, uniform identity, stddev non-negative/zero-for-uniform
- **Marshal round-trip**: generated YAML passes `LoadConfig` + `ValidateConfig`, all services present

### Synth engine (10 tests)

- **Topology conformance**: every span's service.operation exists in the topology
- **Call graph**: every parent-child pair is a valid call in the topology
- **Span nesting**: children start after parent, parent ends after children, durations positive
- **Stats consistency**: exporter span count matches `Stats.Spans`, errors bounded
- **Error cascading**: child errors propagate to parent
- **Span metadata**: root spans are SERVER, non-root CLIENT, all spans share trace ID

### Scenarios (10 tests)

- **ActiveScenarios**: window correctness, sorted by priority, stable ordering
- **ResolveOverrides**: last-defined-wins merge, partial preserves earlier fields, does not mutate input, all refs present
- **ResolveTraffic**: highest priority wins
- **Engine integration**: scenario duration and error rate overrides applied correctly

### Attribute generators (8 tests)

- StaticValue always returns same value, WeightedChoice output in choices, BoolValue output is boolean with correct extremes, RangeValue within bounds and single-value identity, SequenceValue monotonic, NormalValue mean convergence

### Distributions (6 tests)

- Sample non-negative (clamping works), zero stddev returns exact mean, mean convergence, ParseDistribution round-trip

### Topology building (7 tests)

- Refs correct (`service.operation` format, service pointer set), no cycles, roots not called by anyone, roots complete, roots sorted, cycle detection catches deliberate cycles, calls resolved to valid operations

### Rate parsing (4 tests)

- Valid rates parse correctly, zero/negative rejected, exceeds-max rejected, per-second unit correct

### Config validation (5 tests)

- Valid configs accepted, no services rejected, no traffic rate rejected, bad call targets rejected, bad durations rejected

### Traffic patterns (5 tests)

- Uniform constant rate, diurnal bounded between trough and peak, bursty alternation, all patterns non-negative, custom segment boundaries respected

### Semantic conventions (9 tests)

- Registry index consistency (group/attribute lookup), nonexistent returns nil, domains sorted, domain group counts match, merge contains both registries, merge does not mutate originals, ref resolution (type/stability inherited, brief override), load group count matches input

### Circuit breaker state machine (1 test)

- Uses `t.Repeat` to drive random sequences of success/failure requests and time advances against both the real `OperationState` and a simplified model. Checks state agreement (Closed/Open/HalfOpen), rejection behaviour, and active request count invariant after every action

### Fuzz targets (5 targets)

- `FuzzValidateConfig`, `FuzzBuildTopology`, `FuzzParseDistribution`, `FuzzParseRate` in `pkg/synth/`
- `FuzzMarshalRoundTrip` in `pkg/synth/traceimport/`
- Wraps property tests via `rapid.MakeFuzz` for coverage-guided fuzzing with `go test -fuzz`

### Fuzz corpus (197 entries)

- 66 entries for `FuzzMarshalRoundTrip` (from 5-minute run, 521k executions)
- 77 entries for `FuzzValidateConfig`, 20 for `FuzzBuildTopology`, 34 for `FuzzParseRate`
- Stored in `testdata/fuzz/` directories for CI replay
- 2 rapid regression files for bugs found during development (bursty pattern boundary, distribution convergence)

## Rapid features used

- `rapid.Check` — core property testing
- `rapid.Custom` — wraps `genSimpleConfig` for combinator composition
- `rapid.StringMatching` — regex-based generation for durations, rates, error rates
- `rapid.MakeFuzz` — converts property tests to `go test -fuzz` targets
- `t.Repeat` — state machine testing for circuit breaker
- `rapid.Permutation`, `rapid.SampledFrom`, `rapid.Filter` (implicit via construction)

## Bugs found and fixed

1. **Rate overflow**: when traces are nanoseconds apart, computed rate exceeded the 10,000/s validator cap
2. **Fractional rate format**: sub-1/s rates produced `"0.20/s"` which the integer-only rate parser rejected. Now converts to per-minute format

## Test plan

- [x] `make test` passes
- [x] `make lint` passes
- [x] All 81 property tests pass across 3 packages
- [x] All 5 fuzz targets run successfully
- [x] Fuzz corpus replays cleanly during `go test` (no `-fuzz` flag needed)

### CODE REVIEW

File: pkg/synth/property_test.go [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+	if len(topo.Roots) == 0 {
+		t.Skip("no root operations")
+	}
+
+	exporter := tracetest.NewInMemoryExporter()
+	tp := sdktrace.NewTracerProvider(sdktrace.WithSyncer(exporter))
+	defer func() { _ = tp.Shutdown(context.Background()) }()
+
+	seed := genSeed(t)
+	rng := rand.New(rand.NewPCG(seed, 0)) //nolint:gosec // property test
```
[the remark this answers is not in the document]
Comment (andrewh) [reply pkg/synth/property_test.go]: Fixed in 47438aa. Changed all `//nolint:gosec // property test` to `//nolint:gosec // not used for security` across the file.

**Review by andrewh — COMMENTED** [review]
## Self-review

I've read through the full diff. Two issues worth fixing, plus some observations.

### Issues to fix

**1. Variable shadowing in `traceimport/fuzz_test.go:32`**

`f, err := os.CreateTemp(...)` shadows the outer `f *testing.F` parameter. It works by accident because `*os.File` also has a `Name()` method, but it's confusing and would break if we ever needed the `*testing.F` again inside the closure. Should rename to `tmpFile` or similar.

**2. Remaining hardcoded `10000` in `property_test.go:1284` and `1324`**

`TestProperty_ParseRate_ValidRates` and `TestProperty_ParseRate_PerSecondRate` still use `rapid.IntRange(1, 10000)` instead of `MaxRateCount`. Missed these in the constant extraction.

### Observations (no action needed)

- The circuit breaker state machine test is the most valuable single test in the PR — it found real edge cases during development and would catch regressions in the Admit/Enter/Exit state transitions. Good use of `t.Repeat`.
- `genSimpleConfig` only generates DAGs (calls flow to higher-indexed services). This is correct for valid configs but means we never fuzz configs with accidental cycles through this generator. The dedicated `CycleDetected` test covers that path separately, so this is fine.
- The fuzz corpus adds 339 files / ~700 lines. This is reasonable for CI replay speed (subsecond) and the coverage benefit is worth the repo size.
- `genAttribute` in semconv draws from a fixed pool of 5 attribute IDs, which means collisions are common. This is actually good — it exercises the "last writer wins" path in `buildRegistry`.
